xbmc/video/VideoDatabase.cpp

```diff
@@ -2874,13 +2890,49 @@ void CVideoDatabase::ClearBookMarksOfFile(int idFile, CBookmark::EType type /*=
   }
 }
 
+void CVideoDatabase::GetEpisodeByBookmarkID(int bookmarkID, long& season, long& episode)
+{
+  try
+  {
+    if (NULL == m_pDB.get()) return;
+    if (NULL == m_pDS.get()) return;
+
+    std::string strSQL = PrepareSQL("SELECT c%02d, c%02d FROM episode WHERE c%02d=%i ORDER BY c%02d, c%02d", VIDEODB_ID_EPISODE_EPISODE, VIDEODB_ID_EPISODE_SEASON, VIDEODB_ID_EPISODE_BOOKMARK, bookmarkID, VIDEODB_ID_EPISODE_SORTSEASON, VIDEODB_ID_EPISODE_SORTEPISODE);
+    m_pDS->query(strSQL.c_str());
+    if (!m_pDS->eof())
+    {
+      episode = m_pDS->fv(0).get_asInt();
+      season = m_pDS->fv(1).get_asInt();
+    }
+    else
```

Comment:
This whole else block looks unnecessary. The behaviour is exactly the same if you drop it.